client/client_test.go

```diff
@@ -1243,6 +1243,59 @@ func TestClient_UpdateNodeFromDevicesAccumulates(t *testing.T) {
 
 }
 
+func TestClient_UpdateNodeFromFingerprintKeepsConfig(t *testing.T) {
+	t.Parallel()
+
+	// Client without network configured updates to match fingerprint
+	client, cleanup := TestClient(t, nil)
+	defer cleanup()
+	dev := client.config.Node.NodeResources.Networks[0].Device
```

Comment:
It took me a minute to figure out what `dev` was intended to equal here, so perhaps a comment like this would help future readers?

```suggestion
	// Capture fingerprinted network interface for subsequent test
	dev := client.config.Node.NodeResources.Networks[0].Device
```